patches/rosconsole.patch

```diff
@@ -0,0 +1,180 @@
++  if(ANDROID)
++    list(APPEND rosconsole_backend_LIBRARIES rosconsole_android rosconsole_backend_interface log)
++    set(ROSCONSOLE_BACKEND "android")
++  endif()
++endif()
+ if(ROSCONSOLE_BACKEND STREQUAL "" OR ROSCONSOLE_BACKEND STREQUAL "glog")
+   find_package(PkgConfig)
+   pkg_check_modules(GLOG QUIET libglog)
+@@ -85,6 +92,9 @@ elseif(ROSCONSOLE_BACKEND STREQUAL "glog")
+ elseif(ROSCONSOLE_BACKEND STREQUAL "print")
+   add_library(rosconsole_print src/rosconsole/impl/rosconsole_print.cpp)
+   target_link_libraries(rosconsole_print rosconsole_backend_interface)
++elseif(ROSCONSOLE_BACKEND STREQUAL "android")
++  add_library(rosconsole_android src/rosconsole/impl/rosconsole_android.cpp)
++  target_link_libraries(rosconsole_android rosconsole_backend_interface log)
+ else()
+   message(FATAL_ERROR "Unknown rosconsole backend '${ROSCONSOLE_BACKEND}'")
+ endif()
+
+--- /dev/null
++++ catkin_ws/src/ros_comm/rosconsole/src/rosconsole/impl/rosconsole_android.cpp
+@@ -0,0 +1,128 @@
++#include "ros/console.h"
++
++#include <android/log.h>
++
++namespace ros
++{
++namespace console
++{
++namespace impl
++{
++
++std::vector<std::pair<std::string, levels::Level> > rosconsole_android_log_levels;
++LogAppender* rosconsole_android_appender = 0;
++
++void initialize() {}
++
++std::string getName(void* handle);
++
++void print(void* handle, ::ros::console::Level level, const char* str, const char* file, const char* function, int line)
++{
++  // pass log message to appender
++  if(rosconsole_android_appender)
++  {
++    rosconsole_android_appender->log(level, str, file, function, line);
++  }
++
++  android_LogPriority android_log_level;
++  if(level == ::ros::console::levels::Info)
++  {
++    android_log_level = ANDROID_LOG_INFO;
++  }
++  else if(level == ::ros::console::levels::Warn)
++  {
++    android_log_level = ANDROID_LOG_WARN;
++  }
++  else if(level == ::ros::console::levels::Error)
++  {
++    android_log_level = ANDROID_LOG_ERROR;
++  }
++  else if(level == ::ros::console::levels::Fatal)
++  {
++    android_log_level = ANDROID_LOG_FATAL;
++  }
++  else if (level == ::ros::console::levels::Debug)
++  {
++    android_log_level = ANDROID_LOG_DEBUG;
++  }
++  else
++  {
++    // ignore debug
++    return;
++  }
++
++  std::string name = getName(handle);
++
++  std::stringstream ss;
++  ss << name << ": in file:" << file << ":" << line;
++  __android_log_print(android_log_level, ss.str().c_str(), "%s", str);
++}
++
++bool isEnabledFor(void* handle, ::ros::console::Level level)
++{
++  size_t index = (size_t)handle;
++  if(index < rosconsole_android_log_levels.size())
++  {
++    return level >= rosconsole_android_log_levels[index].second;
++  }
++  return false;
++}
++
++void* getHandle(const std::string& name)
++{
++  size_t count = rosconsole_android_log_levels.size();
++  for(size_t index = 0; index < count; index++)
++  {
++    if(name == rosconsole_android_log_levels[index].first)
++    {
++      return (void*)index;
```

Comment:
Is this returned `handle` stable? Probably yes, because elements are never removed after they have been added to the vector.